overlord/restart/restart_parameters.go

```diff
@@ -0,0 +1,51 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2023 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package restart
+
+import (
+	"github.com/snapcore/snapd/boot"
+	"github.com/snapcore/snapd/bootloader"
+)
+
+type RestartParameters struct {
+	SnapName          string              `json:"snap-name,omitempty"`
+	RestartType       RestartType         `json:"restart-type,omitempty"`
+	BootloaderOptions *bootloader.Options `json:"bootloader-options,omitempty"`
+}
+
+func (rt *RestartParameters) init(snapName string, restartType RestartType, rebootInfo *boot.RebootInfo) {
+	// restart types of type *Now takes precedence
+	switch restartType {
+	case RestartSystemNow, RestartSystemHaltNow, RestartSystemPoweroffNow:
```

Comment:
as we discussed, do we need a precedence order also between the operations RestartSystem vs Halt vs Powerwoff ?